pinot-common/src/main/proto/plan.proto

```diff
@@ -0,0 +1,74 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+syntax = "proto3";
+
+package org.apache.pinot.common.proto;
+
+message StageNode {
+  int32 stageId = 1;
+  string nodeName = 2;
+  repeated StageNode inputs = 3;
+  ObjectFields fields = 4;
+}
+
+message ObjectFields {
+  map<string, LiteralField> literalField = 1;
+  map<string, ListField> listFields = 2;
+  map<string, MapField> mapFields = 3;
+}
+
+message LiteralField {
+  oneof literal_field {
+    bool boolField = 1;
+    int32 intField = 2;
+    int64 longField = 3;
+    double doubleField = 4;
+    string stringField = 5;
+    bytes bytesField = 6;
+  }
+}
+
+message ListField {
+  repeated ObjectFields objects = 1;
+  repeated LiteralField literals = 2;
+}
+
+message MapField {
+  map<string, ObjectFields> objects = 1;
+  map<string, ObjectFields> literals = 2;
```

Comment:
same here

Reply:
Also is MapField specifically for the purpose where a plan node has a map / hashmap ?